### PR TITLE
Improve styles for dragging and dropping files

### PR DESCRIPTION
**Pull request by flamerohr**
Also added a "shift+click" function, so you can select two files while holding shift and all files between them will also be selected

screen shot 2018-02-20 at 11 02 33 am

Please see https://github.com/silverstripe/silverstripe-asset-admin/issues/424 for more details.

cc: @newleeland for design review

### CODE REVIEW

File: client/src/components/GalleryItem/GalleryItemDragLayer.js [hunk 1]
```diff
@@ -19,23 +22,31 @@ class GalleryItemDragLayer extends Component {
     if (!item.selected) {
```
Comment (unclecheese) [inline client/src/components/GalleryItem/GalleryItemDragLayer.js]: Could use `const { item } = this.props` just for consistency.

File: client/src/components/GalleryItem/GalleryItemDragLayer.js [hunk 2]
```diff
@@ -19,23 +22,31 @@ class GalleryItemDragLayer extends Component {
...
       return null;
     }
-    const selected = `${item.selected.length}`;
+    const shadows = [
+      item.selected.length > 1
```
Comment (unclecheese) [inline client/src/components/GalleryItem/GalleryItemDragLayer.js]: `item.selected.length` has several references in this function. Maybe assign it to `selectionCount` just to keep things tidy?

**Comment by unclecheese** [comment]
Looks great. I just made some minor changes to variable names and assignments. Overall really nice work on a pretty gnarly API!


Biggest issue is browser testing. All versions of IE and Edge are having issues with the drag layer. Basically, the native HTML5 drag layer is shown, but your custom folder designs are hidden behind it.

It doesn't completely kill the UX, but it does render most of this work moot for IE users, so might want to ask PO @chillu how much time you want to invest in fixing it.

dnd

**Comment by unclecheese** [comment]
I remember having to navigate this on [kickassets](github.com/unclecheese/silverstripe-kickassets), and I believe this was the solution I used https://stackoverflow.com/questions/36379184/html5-draggable-hide-original-element#answer-36379727

**Comment by chillu** [comment]
IE is still a sizeable portion of users. Google Analytics says around 8% of traffic on userhelp. I'd cap investigation into this at 2h, otherwise we'll leave it.

**Comment by unclecheese** [comment]
The `git diff` NPM test is failing. Will rebuild and push up and see if that sorts it.

**Pull request merged**